modules/brainbrowser/ajax/minc.php

```diff
@@ -22,13 +22,56 @@
 
 $headers = array();
 
-$query     = "select File from files where FileID = :MincID";
-$minc_file = $DB->pselectOne($query, array('MincID' => $_REQUEST['minc_id']));
-$minc_file = getMincLocation() . $minc_file;
+if (strpos(escapeshellcmd($_REQUEST['minc_id']), 'l') !== false) {
+    list($l, $id) = explode('l', escapeshellcmd($_REQUEST['minc_id']));
```

Comment:
Why are these being shell escaped here? It's not being passed to a shell or used as a command. This should be done right before the `passthru` command, not everywhere that the minc_id is used.

If you really want to check if the user was trying to inject something and abort early, you could probably do something like `if (escapeshellcmd(x) !== x) { print error and exit} ` (note: I haven't tried that, but it should work.)

There's also this warning from the escapeshellcmd documentation that should be taken into account:

`Warning escapeshellcmd() should be used on the whole command string, and it still allows the attacker to pass arbitrary number of arguments. For escaping a single argument escapeshellarg() should be used instead.` (http://php.net/manual/en/function.escapeshellcmd.php)